internal/server/grants.go

```diff
@@ -14,6 +14,25 @@ import (
 	"github.com/infrahq/infra/uid"
 )
 
+type requestID interface {
+	RequestID() uid.ID
+}
+
+var listGrantsRoute = route[api.ListGrantsRequest, *api.ListResponse[api.Grant]]{
+	authorization: func(rCtx access.RequestContext, req any) (ac.PermissionSet, error) {
+		if reqID, ok := req.(requestID); ok {
+			if reqID.RequestID() == rCtx.Authenticated.User.ID {
+				// TODO: how does permission set work in this case?
+				return nil, nil
+			}
+
+			// TODO: user in group
```

Comment:
These are the last problems to solve for this approach I think. If the data layer (and other lower level packages) need to check the permissions then I think the `PermissionSet` needs to include an ID. That's easy enough for the "owned by user" case. The "owned by group" case is more involved. The permissions might need to be functions instead of simply an ID.